### DISCUSSION TITLE
Import GeoJSON points, lines, polygons in favorites

### DISCUSSION THREAD

**Discussion by DIrgi** [devemux86/cruiser#506] [Q&A] [answered]
![image](https://github.com/devemux86/cruiser/assets/57968526/ddb5485e-ee7e-46fc-bddc-552cce2042c9)

![image](https://github.com/devemux86/cruiser/assets/57968526/517490d8-22d2-4aff-9b88-73cdd09cffb1)

I "played" a little bit with overpass-turbo and extracted all french departments (still a draft) to display  "low_emission_zones"
catchword ["Crit'Air"](https://www.certificat-air.gouv.fr/) as a first starter ( I know other countries have zones like this as well but this can/will be step II).

I found a similar discussion here, dealing with [BRouter NoGo areas](https://github.com/devemux86/cruiser/discussions/85). But finally it's not the same. The "No go" areas deal with what the name says, defined zone where the BRouter algorithm should plan a bypass.
Well finally a low emission zone is something similar in the end. But I don't want to go that deep into it because if so, it requires life data to tell you if and when the zone is temporary active for whom and so on.
However for me it's just fine to see this area as a transparent polygon overlay on the map because depending on your vehicle it's finally not allowed to enter no matter if there's an alarm or not.

**Coming to Cruiser:**
I tried to install it in every possible way into Cruiser but finally there's not really a solution for this so far (at least I didn't find it). Even when I load it in as overlay, the maximum I received was like seeing the area border as a kind of a surrounding track but not just as a informal transparent polygon area. See the screenshot picture from overpass to see what I mean.

![image](https://github.com/devemux86/cruiser/assets/57968526/e6e09bea-336e-4414-b867-306dcf7b362d)
![image](https://github.com/devemux86/cruiser/assets/57968526/ebb2a71b-80bc-46e0-bf10-0adc761e4548)

Cruiser offers the possibility to import own pois and tracks. Can we have the same solution for polygon area overlays? And then to select the color and transparency? - that would be great.

For a first test, I attached the data from the map above as gpx file (zipped) and as original [Overpass-Turbo](https://overpass-turbo.eu/) query content for copy/paste to reproduce request if needed.

[overpass query request.txt](https://github.com/devemux86/cruiser/files/15053201/overpass.query.request.txt)
[Crit'Air.zip](https://github.com/devemux86/cruiser/files/15053268/Crit.Air.zip)

**Comment by devemux86** [comment]
@DIrgi thanks for the suggestion.

Overpass can export in GeoJSON format?
GeoJSON is the appropriate format for storing all kinds of geometries and polygons (also with holes).

Cruiser can display GeoJSON content, it's just not public yet, unless we need it. :slightly_smiling_face:

**Reply by DIrgi** [reply]
Yes it can export geojson but it does not work neither as track nor as route.
The extracted file has round about 80MB. Check by yourself, I attach it.
[export.zip](https://github.com/devemux86/cruiser/files/15053350/export.zip)

**Reply by devemux86** [reply]
As mentioned, the GeoJSON format is not currently public in Cruiser.

We can think how it could be used, e.g. in "My collection" requires a new implementation.

For now can use GPX tracks and draw them as lines.

**Reply by DIrgi** [reply]
"My collection" makes absolutly sense. 

Like a track, POI or Route. Then maybe an "area" just as overlay (without any routing influence - just informal)
with own defined colors and transparency, switch on/off in settings on demand.

ECO zones are coming more and more - See screenshot for Europe - Even in the OSM data they are not 100% listed.
![image](https://github.com/devemux86/cruiser/assets/57968526/575e7db0-778d-4f88-a1ee-cee78d389e53)


i.e. I started the tread with an example picture about french department zones. I check via JOSM but in France there are no "low emmission zone" definitions so far for whole departments (ZPAd) , just for certain city areas(ZFE). But since these emission zones affect a complete department, I asked for the department borders to receive the outcome for the low emission department zones.
That these zones are existing for complete departments can be read on the official government page - See link crit'air, where the city zones are ZFE and the department zones are the ZPAd.

**Comment by devemux86** [comment]
An old image:

![cruiser](https://github.com/cgeo/cgeo/assets/3484020/e771e3c9-8e86-470b-935c-e59b36642e52)

**Comment by devemux86** [comment]
Displaying polygon overlays is one thing.

Store them like favorites / routes / tracks with adjustable properties,
this requires more work and a new menu category in "My collection"?

**Comment by DIrgi** [comment]
Will we see GeoJSON format in Cruiser 5 ?

**Reply by devemux86** [reply]
It will be available, we will see when, as many things are in the works all the time. :slightly_smiling_face:

**Comment by devemux86** [comment]
Next Cruiser will also import GeoJSON files:
- GeoJSON points into "Favorites"

- GeoJSON lines into "Tracks"

- GeoJSON polygons into **?**

We can rename "My collection | Tracks" to "Tracks / Areas" and store 2 overlay types:
- Lines
- Polygons (with a different icon)

**Reply by devemux86** [reply]
Remember that Cruiser can already import GeoJSON routes from BRouter website,
_offline without route calculation_ with "Import" menu or into "My collection | Routes".

**Reply by bios4** [reply]
> GeoJSON polygons into ?

...a new section called "Areas"!?
Why mix it with "Tracks"?

**Reply by DIrgi** [reply]
Or zones but area is on a more broader scale

**Reply by devemux86** [reply]
It's better to not introduce a new interface just for a different type,
which can only come from one format and its uses are still in doubt.

We can always create different groups to separate the items.

There are the main categories "Favorites" and "Routes" with specific uses
and practically everything else.

**Reply by bios4** [reply]
So why ask then when you don't want to introduce a new category at all?

I even don't get why it should be "better to not introduce a new interface just for a different type"...
I also wonder what the icon shall then look like for "Tracks/Areas", but hey.

I'm sure you will figure something out.

**Reply by devemux86** [reply]
The question was asked 2 days ago, work has progressed a lot since then.

The icon is the smallest part of the work, the implementation (and maintenance) has a much higher cost.

**Comment by MotoUKRider** [comment]
I have a (slightly modified) [Polyfile creator (using Leaflet) here](https://www.thenoddingbiker.com/polyfile/) for creating custom map areas, turns out that GeoJSON is also offered so I can conjure-up all sorts of fancy areas/zones where a simple radius may not be sufficient.

For starters, the London Congestion Charge Zone:
*(it's a very busy list, so the JSON is outside of the screenshot!)*

![congestion-zone-polyfile](https://github.com/devemux86/cruiser/assets/125696109/9a48dce4-d27f-4aae-9ff0-d648eebb57d8)

GeoJSON content:
[london-congestion-charge-zone.txt](https://github.com/user-attachments/files/16161988/london-congestion-charge-zone.txt)

**Reply by devemux86** [reply]
Complicated shapes are expensive in memory, rendering, routing, storage.

It is better to prefer simple shapes: points, lines, circles that are optimized.

**Reply by WolfgangZ42** [reply]
For the Munich Diesel-Ban-Area I created last year a simple shape: (see https://github.com/devemux86/cruiser/discussions/85#discussioncomment-5577744)
![image](https://github.com/devemux86/cruiser/assets/121806238/36d8a212-4c34-46d0-bacd-3ea48dfb64ef)
works well - but it's not displayed.
Now I take the track around the Diesel-Ban-Area and could see it as overlay:
![2024-07-10 16 59 32](https://github.com/devemux86/cruiser/assets/121806238/1b6c6e34-b972-4853-9d8f-4cb949f89f74)
I know, it's not a "real" no-go-area, just the streets around are nogos, but it works.
Both are simple gpx files.

**Reply by devemux86** [reply]
This quickly becomes complicated. GPX are designed to host points and lines.
Also routing with line geometries as avoid areas will have performance issues.

And there is no way for the users to design or find these shapes,
as it is expected everything to be handled with the app interface.

The best is to follow BRouter website's model and interface and use avoid circles.
We could use the favorites: groups having a flag to host no-go points with radius.

**Reply by devemux86** [reply]
About the favorite no-go areas we have this thread:
- https://github.com/devemux86/cruiser/discussions/545

**Reply by DIrgi** [reply]
Radius areas might be easy to define but a zone is never radius based. Finally with a circle you either add areas which are then not included or vice versa, which are normally out of it

**Reply by 0709wiwiwi** [reply]
> For starters, the London Congestion Charge Zone.

Q:  @MotoUKRider  
Did you manually click and place all those points ? 
Or was a file already publicly available somewhere ?

**Reply by MotoUKRider** [reply]
@0709wiwiwi I might need a new mouse!  I had nothing else better to do, so made it manually using my Polyfile webpage by following [the reasonably high-quality map](https://ulez.co.uk/wp-content/uploads/2023/09/Congestion-Charge-Zone-Map.png) that the Transport for London offer.

**Comment by devemux86** [comment]
GeoJSON **area** overlays will be available in the next Cruiser.

- They will use a highly optimized format.
The [above](https://github.com/devemux86/cruiser/discussions/506#discussioncomment-9181435) France example is 75 MB, but it will be only 2.5 MB in Cruiser.

- Currently there is no hole support.
I'm not sure if we need it? Not to complicate things.

- Do we need any extra info in the areas list?
For tracks we have length, duration, ascend, descend.

<img src="https://github.com/user-attachments/assets/b1915458-99c6-4a45-85c1-608f29d6f69d" width="33%"/><img src="https://github.com/user-attachments/assets/c63cf84d-7de8-4727-a4c9-be15c251d423" width="33%"/><img src="https://github.com/user-attachments/assets/1678dbd1-5043-44cc-b35f-87be3626fc25" width="33%"/>

**Reply by devemux86** [reply]
The nogo_Munich_MR example by @WolfgangZ42 as a "closed" gpx track:

[nogo_Munich_MR-track.gpx.txt](https://github.com/user-attachments/files/16201400/nogo_Munich_MR-track.gpx.txt)

<img src="https://github.com/user-attachments/assets/fa053ac8-9143-44fd-ae5c-330f23df6691" width="50%"/><img src="https://github.com/user-attachments/assets/85abe19f-906e-44da-8e39-3b2747a79638" width="50%"/>

**Reply by devemux86** [reply]
@WolfgangZ42 

Does this gpx work as no-go in BRouter routing?

(it contains track instead of route)

**Reply by DIrgi** [reply]
Finally a zone should support two things either just visual and/or working as a „no go“ zone for the navigation.

if especially area blocking is to complicated,
Visual should work at least, so you can see while planning which areas you should avoid if needed.

having influence on routing automatism might be much more complicated ‚cause the the algorithm need to know not to navigate there on demand and it’s not just like a closed/blocked road it has to block every action in this area.

**Reply by devemux86** [reply]
Navigation uses the calculated route, so it is about the route calculation.

Using complex shapes as no-go areas is not simple for the routers, we will see if and what is possible.
No-go circles is certainly the easiest as demonstrated and used in BRouter website.

**Reply by WolfgangZ42** [reply]
> Does this gpx work as no-go in BRouter routing?

As I wrote three days ago (https://github.com/devemux86/cruiser/discussions/506#discussioncomment-10011479) in Cruiser it works. Since more as a year "in production"! For visual support only I now use [nogo_Munich_MR_track.gpx.txt](https://github.com/user-attachments/files/16201011/nogo_Munich_MR_track.gpx.txt), works as well. Btw: it's a navigatable track!
Please keep the possibility to specify no-go-areas in /Routing/BRouter(offline)/Advanced - profile related. 

The website of BRouter expects .geojson files for no-go-areas. Transfering my navigatable track to .geojson was quite easy and now I've an exact no-go-area of the Munich-Diesel-Ban in BRouter as well:

w/o no-go-areas:
![2024-07-13 08_23_39-BRouter Web Client](https://github.com/user-attachments/assets/c1733200-71a6-4964-8da8-efc6e4ee08c0)

loading the no-go-area:
![2024-07-13 08_25_12-BRouter Web Client](https://github.com/user-attachments/assets/7328ec9b-8959-4df8-b125-c3b5e40541aa)

here we go:
![2024-07-13 08_25_26-BRouter Web Client](https://github.com/user-attachments/assets/eb3a00ad-7cbe-44a9-8103-592121cc134a)

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the info. :+1: 

> Please keep the possibility to specify no-go-areas in /Routing/BRouter(offline)/Advanced - profile related.

Adding no-go areas in favorites, we do not need external no-go areas that complicate the implementation.
(we should really try to simplify Cruiser and maintentance :slightly_smiling_face:)

**Reply by devemux86** [reply]
Cruiser will also read GPX tracks and routes,
automatically detect if their geometry is closed and can be imported as areas.

For example the no-go area by @WolfgangZ42 as track and route:
[nogo_Munich_MR-track.gpx.txt](https://github.com/user-attachments/files/16201401/nogo_Munich_MR-track.gpx.txt)
[nogo_Munich_MR-route.gpx.txt](https://github.com/user-attachments/files/16201402/nogo_Munich_MR-route.gpx.txt)

**Reply by Ciutnik** [reply]
What actually determines that Cruiser will consider a path a closed area? Is it enough that the first and last points have the same coordinates? @WolfgangZ42's zone contains the `x:closed` tag, is this necessary?

**Reply by devemux86** [reply]
> @WolfgangZ42's zone contains the x:closed tag, is this necessary?

This was a test, I updated the attached gpx.

> Is it enough that the first and last points have the same coordinates?

This should be enough for most purposes (routes or tracks).

**Reply by Ciutnik** [reply]
Since the beginning of July, the NoGo zone has also been introduced in my city. Unfortunately, it covered me both as the owner of the banned car and as a resident of this zone! It's good that due to where I live, my entry ban is deferred for 4 years...

That's why I wanted to try NoGo zone support in Cruiser (4.1.0-b2). I know that a new version will be released soon with completely new support for traces/areas, but with the latest beta I am not able to use my NoGo zone.

Firstly, offline BRouter calculations do not respect this NoGo zone, and secondly, it is not remembered in the application. Maybe the file should be placed in the application's data folder?

I thought it was because I had drawn the zone in Locus, which added some of its own extensions, but after importing the trace into Cruiser and then exporting it, nothing changed.

[NoGo_SCT-Warszawa.gpx.txt](https://github.com/user-attachments/files/16202500/NoGo_SCT-Warszawa.gpx.txt)

**Reply by devemux86** [reply]
@Ciutnik

External no-go files will indeed be removed as they are complicated to maintain.
Everything will be handled inside the favorites.

@WolfgangZ42 mentions [above](https://github.com/devemux86/cruiser/discussions/506#discussioncomment-10037487) that uses properly external no-go files.

Currently, just need to open them in BRouter advanced settings.

**Reply by Ciutnik** [reply]
> @WolfgangZ42 mentions above that uses properly external no-go files.

That's it! That's why I was interested in it, but without success... Well, since the implementation will change completely soon anyway, it's not worth bothering with it now. I will definitely come back to this in the new version of Cruiser 🙂

**Reply by devemux86** [reply]
Testing the nogo_Munich_MR-track.gpx from [above](https://github.com/devemux86/cruiser/discussions/506#discussioncomment-10033197) I see that it works (and remembered) correctly.

Perhaps need to double check the format of your gpx?

<img src="https://github.com/user-attachments/assets/2bf39b24-de74-4a3e-a4b0-1adefd71cf65" width="33%"/><img src="https://github.com/user-attachments/assets/65473a2d-d641-470c-831f-b41b6bf2ff58" width="33%"/><img src="https://github.com/user-attachments/assets/681cc5f3-9252-4804-a442-bad133dcf04d" width="33%"/>

**Reply by Ciutnik** [reply]
> Perhaps need to double check the format of your gpx?

I attached my file earlier. This is GPX exported from Cruiser, then (failed) loaded as No-Go, so it's probably correct. Compared to Wolfgang's file, it has more points and no extension `x:closed` (that's why I asked about it earlier).

**Reply by WolfgangZ42** [reply]
> **Please keep the possibility to specify no-go-areas** _in /Routing/BRouter(offline)/Advanced -_ **profile related.**

The last two words are important. Nothing against a no-go-solution via "My collection" as long as different definitions for no-go-areas per profile are possible. I have already written about this recently: https://github.com/devemux86/cruiser/discussions/545#discussioncomment-9530310
An excerpt from it:
Imagine you define no-go-areas for the bycicle profile (e.g. stairways) you don't need in the walking profile. Or you'll avoid inner cities around your home address permanently, but only for motorcycle tours.

**Reply by Ciutnik** [reply]
@WolfgangZ42 I completely agree with you that zones must be defined independently for each profile! This is obvious. The zone in my city applies to some cars (including mine) but not to any motorcycles.

**Reply by devemux86** [reply]
Favorites are global, it is impossible each created profile to have different favorites.
Imagine the complexity in the interface, implementation and storage... :slightly_smiling_face: 

We can have groups for different travel modes and enable / disable them.

e.g. fuel station favorites are not needed in walking profiles

**Reply by Ciutnik** [reply]
@devemux86 

> We can have groups for different travel modes and enable / disable them.

So, if I understand correctly, I will be able to permanently turn on the no-go zone stored in the library for the "Car" profile, but turn it off for "Motorcycle"?

**Reply by devemux86** [reply]
Favorite no-go are like all other favorites.

As we can set the nearby favorites to be used in navigation or not,
we can set the no-go points / circles to be used in routing or not.

What @WolfgangZ42 describes is an automatic way, but this is complicated.
We can think about it.

**Reply by 0709wiwiwi** [reply]
@Ciutnik 
Produce the closed "Munich" nogo gpx track in no time with Cruiser.
Then place this nogo gpx track file in your DIESEL custom car profile and you're done. 
Make sure not to accidentally change this DIESEL car profile afterwards. No warning !

**Reply by Ciutnik** [reply]
@0709wiwiwi 
I would like to do this with my nogo track "Warszawa", but nothing happens. BRouter offline does not respect this by calculating the route through the center of the zone, and Cruiser does not remember the zone settings.

**Reply by devemux86** [reply]
@Ciutnik if you have another device, can you try with `4.0.2`?

**Reply by Ciutnik** [reply]
It's the same on 4.0.2. I load GPX as No-Go, restart Cruiser, and No-Go areas is "None". Maybe there's something wrong with my GPX? ~~Maybe a space in the name?~~ Or something like that?

**Reply by devemux86** [reply]
@Ciutnik probably, as it works for other users all this time.

Try the Munich gpx by @WolfgangZ42 that we tested earlier and it works.
Can also change the gpx location from where you open it.

**Reply by Ciutnik** [reply]
Interesting thing. The loaded "Munich" is remembered after Cruiser restart! Probably, as expected, navigation is not carried out here either.

**Reply by 0709wiwiwi** [reply]
I had no much time now to test...but I know my Munich file did work.  
Try  pse again using nogo   (no capitals) NOT NoGo

**Reply by Ciutnik** [reply]
> Can also change the gpx location from where you open it.

I tried. It doesn't matter.

Could the fact that I am currently inside the zone be important?

**Reply by Ciutnik** [reply]
> Try pse again using nogo (no capitals) NOT NoGo

Do you think such a detail could be important?... I'll try.

EDIT: Bingo! @0709wiwiwi Well done in! That was it! Changing "NoGo" to "nogo" helped. Thanks 😁

**Reply by devemux86** [reply]
> Could the fact that I am currently inside the zone be important?

@Ciutnik you mentioned that the file cannot be read / remembered.
This must be solved first.

If it can be read, then it will be used in routing.

**Reply by devemux86** [reply]
@Ciutnik I tested quickly your file and BRouter does not like it. :slightly_smiling_face: 

I don't know why, the file by @WolfgangZ42 works, it is a route.
And it works even if we convert it to track.

Anyway this proves that external files cannot be trusted and Android does not like external files too.
Let's see what can be done with the favorites that are controlled by the app.

**Reply by Ciutnik** [reply]
@0709wiwiwi suggested changing the "NoGo" to "nogo" in the name and that was it! Now it works OK 👍 I wasn't aware of case sensitivity. Thanks for looking into the problem 🙂

**Reply by devemux86** [reply]
@Ciutnik really, it wants lowecase?

This is problem of Android and how it handles the external file names.
If it has valid characters (and it has) it should work.

**Reply by 0709wiwiwi** [reply]
short ...I suppose (should read the documentation) that BRouter expect it this way as non Capitalised.
By the way the routing can escape out of nogo zones but can not not enter them is same in Locus. .is expected behaviour ;-).

**Reply by devemux86** [reply]
BRouter no-go documentation is here:

https://github.com/abrensch/brouter/blob/master/docs/users/android_advanced.md#using-nogo-areas

**Reply by 0709wiwiwi** [reply]
> this proves that external files cannot be trusted.

This tale of doom does come back very often. 
File transfer support are like oxygen for Cruiser.
Without oxygen, Cruiser can not survive.

**Comment by WolfgangZ42** [comment]
The fact that Android tightens the security restrictions from release to release is understandable on the one hand, but annoying on the other. The restrictions for external files may also become critical, but Import/Export will still have to exist. So external file handling has to be there anyway. 

But nevertheless to minimize the use of external files, the nogo areas in "My collection" - okay! All "My collection" items are handled globally, thus all nogo-areas would then be applied in all profiles. That's sub optimal.
But couldn't we define different nogo-folders and activate them via /Routing/Options/Advanced - individually for each profile? The .gpx files are then loaded not externally but out of "My collection". Default could be e.g. My Collection/Tracks+Areas/nogo_areas and if no differentiation is needed you don't have to change anything. 
By the way, I am currently already working according to this: the file nogo_Munich_MR_track(.gpx) is available in My Collection/Tracks/nogo_areas (for display) and as an external file for /routing/options/advanced (for routing).

![2024-07-14 23 31 29](https://github.com/user-attachments/assets/bbd2cb2c-cb77-48a5-9246-d1f2aba6a507) . . . ![2024-07-14 23 08 32](https://github.com/user-attachments/assets/16842f2c-675c-4bda-83b5-351a1227affc)

I accept the objection that this idea increases the complexity of Cruiser. But the topic of nogo areas has already been placed in the "Advanced" section. And we can see what leveling down to mediocrity feels like at K3 ...

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the detailed description.

Indeed, we could improve the use of all favorite types with the profiles.

I still have to think about an optimal implementation. :slightly_smiling_face:

**Comment by devemux86** [comment]
Import GeoJSON points, lines, polygons in favorites is available in:

- https://github.com/devemux86/cruiser/discussions/610

- "Closed" GPX tracks / routes can also be imported as polygons

- Polygons can be used as [no-go areas](https://github.com/devemux86/cruiser/discussions/538) in route planning

**Reply by DIrgi** [reply]
I'll test.
Question: Can "no go areas" just be "visible only" without any influence on the routing calculation - just for informal reasons?
Thinking about the environmental zones where it might be to complicated since additional daily live information are needed.

**Reply by devemux86** [reply]
> Can "no go areas" just be "visible only" without any influence on the routing calculation - just for informal reasons?

They can as the visibility and the no-go flag are set separately.

**Comment by devemux86** [comment]
Please do not set many and complex polygons as no-go areas.

This can significantly slow down the route calculation.

**Comment by DIrgi** [comment]
**Test wit New Cruiser APK: cruiser-5.0.0-b1**

First of all it seems that I can import polygon areas to mark/display the french environment zones.
I just want them to be displayed and not as no-go-area - so I don't know at this point if a no-go area will work for planning as well.
Never the less.

![image](https://github.com/user-attachments/assets/571e4d79-5401-461e-aa51-72448b5700b0)


Question: will it be possible to have this as an hatched area on demand in all 4 directions like this:
(just an example)
![image](https://github.com/user-attachments/assets/4827e4fe-fd59-4c5e-b31b-c6b788afe785)

**Reply by DIrgi** [reply]
btw if someone need to have the french crit'air zones:
Here they are, ready for direct import:
[Crit'Air.zip](https://github.com/user-attachments/files/16499700/Crit.Air.zip)

**Reply by devemux86** [reply]
> I just want them to be displayed and not as no-go-area

Import them only, make them visible, but do not set them as no-go areas.
Then they will be used only as overlays.

> I don't know at this point if a no-go area will work for planning as well

I tried to all these France polygons as no-go areas and routing became too slow.

If want to use them as no-go areas, you need to import them as separate layers
and enable only the no-go areas you need in your route, not the whole France!

> Question: will it be possible to have this as an hatched area on demand in all 4 directions like this

Do you mean a pattern instead of the semi-transparent fill color of the polygons?
With the new map engine, such things are more easily possible. :slightly_smiling_face:

**Reply by DIrgi** [reply]
yes, "pattern" English translation named it as "hatched". Finally I'm not aware how it'll look like at the end maybe it's to strong to disturbing....
It has to be checked but as long as it's not  there it cannot be checked. At least the transparency grade should be adjustable and probably the distance between the hatching lines or the amount of has to be in a good relationship otherwise it's probably to "strong" but a hatched or and area with a pattern helps to display that it's an "no go area" or a special zone. 
i.e. If I make it green it can be easily mixed up with forest, gras and so one.

**Comment by devemux86** [comment, marked as answer]
- Import GeoJSON points, lines, polygons in favorites is available in https://github.com/devemux86/cruiser/discussions/641

  - "Closed" GPX tracks / routes can also be imported as polygons

  - Polygons can be used as [no-go areas](https://github.com/devemux86/cruiser/discussions/538) in route planning
